scripts/scheduler_ingest.py

```diff
@@ -0,0 +1,213 @@
+        :param raw_maintenance_data: Maintenance day data received from the Scheduler API
+                                     (as a list)
+        :return: A pre-processed cycle data list, and pre-processed maintenance data list """
+        cycle_data = self._sort_by_date(raw_cycle_data)
+        cycle_data = self._clean_data(cycle_data)
+
+        maintenance_data = self._sort_by_date(raw_maintenance_data)
+        maintenance_data = self._clean_data(maintenance_data)
+
+        return cycle_data, maintenance_data
+
+    def _clean_data(self, data):
+        """ Removes invalid items from the data.
+        :param data: The list of items to be validated
+        :return: The list of items that were passed in, but with all invalid items removed """
+        print("\nBEGIN Data clean")
+        print(f"Original length: {len(data)}")
+        clean_list = []
+        for item in data:
+            if 'name' in item and not re.search(self._cycle_name_regex, item['name']):
+                print(f"\nItem removed due to strange cycle name "
+                      f"({item['name']}).\nFull item: {item}")
+            elif item['start'].replace(tzinfo=None) < self._earliest_possible_date:
+                print(f"\nItem removed due to impossible date "
+                      f"({item['start']}).\nFull item: {item}")
+            else:
+                clean_list.append(item)
+        print(f"New length: {len(clean_list)}")
+        print("END Data clean\n")
+        return clean_list
+
+    @staticmethod
+    def _combine_lists(first, second):
+        """ Combines two lists together.
+        :param first: The first list to be combined
+        :param second: The second list to be combined
+        :return: The single combined list """
+        combined = first + second
+        return combined
+
+    def _sort_by_date(self, data):
+        """ Sorts a list by a date field (specified by the _sort_by_field variable)
+        :param data: The list to be sorted
+        :return: The list sorted by the specified field """
+        return sorted(data, key=lambda date: date[self._sort_by_field])
+
+    @staticmethod
+    def _process_from_separate(cycle_data, maintenance_data):
+        """ Converts pre-processed cycle data items and maintenance day data items
+        into a single list of Cycle objects. Each cycle object contains a list
+        of MaintenanceDay objects - all of which fall within the given cycle period.
+        Maintenance Days which fall outside of cycle periods are highlighted and discarded.
+        :param cycle_data: A pre-processed list containing all valid cycle data items.
+        :param maintenance_data: A pre-processed list of all valid maintenance day data items.
+        :return: A list of Cycle objects containing a list of 0 or more MaintenanceDay objects. """
+        cycle_list = []
+        unused_maintenance_data = maintenance_data.copy()
```

Comment:
`unused_maintenance_data` might be better as just `maintenance_data`. The `unused` suggests that they are not going to be used at all (atleast that's how I read it)

Reply:
For readability, I feel inclined to avoid using the same name as the method argument `maintenance_data` (the method which this variable is a local copy of).

Would `maintenance_data_copy` work perhaps?